stdlib/domain.ml

```diff
@@ -172,17 +87,17 @@ let do_before_first_spawn () =
     first_spawn_function := (fun () -> ())
   end
 
-let at_exit_key = DLS.new_key (fun () -> (fun () -> ()))
+let at_exit_key = CamlinternalTLS.new_key (fun () -> (fun () -> ()))
```

Comment:
Given that the installation of `Domain.at_exit` callbacks is not a performance-sensitive operation, it seems easiest to have a global, mutex-protected hash table indexed by the domain id.

Reply:
Implemented in https://github.com/ocaml/ocaml/pull/12724/commits/2b962452ee4f2b5947016386b1ff89143a2b94a6. The commit uses `Map` instead of `Hashtbl` to avoid a dependency cycle.